### PR TITLE
SED-3352 ap executions filter by plan categories

### PR DESCRIPTION
**Pull request by david-stephan**

### CODE REVIEW

**Review by jeromecomte — COMMENTED** [review]
PR looks good. Just having some questions

File: step-automation-packages/step-automation-packages-junit/src/main/java/step/automation/packages/junit/IncludePlanNames.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/***
+ * Specify automation package plans to run
+ * 
+ * @author exense team
+ */
+@Retention(RetentionPolicy.RUNTIME)
+public @interface IncludePlanNames {
```
[the remark this answers is not in the document]
Comment (david-stephan) [reply step-automation-packages/step-automation-packages-junit/src/main/java/step/automation/packages/junit/IncludePlanNames.java:30]: Rename include and exclude plans

File: step-cli/step-cli-launcher/src/test/java/step/cli/StepConsoleTest.java [hunk 2]
```diff
@@ -125,6 +125,7 @@ public void testRemoteExecute() {
         // all parameters
         int res = runMain(histories, "ap", "execute", "-p=src/test/resources/samples/step-automation-packages-sample1.jar",
                 "-u=http://localhost:8080", "--projectName=testProject", "--token=abc", "--async", "--includePlans=p1,p2",
+                "--includeCategories=CatA,CatB", "--excludeCategories=CatC,CatD",
```
[the remark this answers is not in the document]
Comment (david-stephan) [reply step-cli/step-cli-launcher/src/test/java/step/cli/StepConsoleTest.java:128]: please see my answer for related above comments

File: step-automation-packages/step-automation-packages-junit/src/test/java/step/automation/packages/junit/StepAutomationPackageRunWithCategories.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package step.automation.packages.junit;
+
+import org.junit.runner.RunWith;
+import step.junit.runner.Step;
+
+@RunWith(Step.class)
+@IncludePlanCategories(value = {"My Category A", "My Category B"})
+@ExcludePlanCategories(value = {"My Category C", "My Category D"})
```
[the remark this answers is not in the document]
Comment (david-stephan) [reply step-automation-packages/step-automation-packages-junit/src/test/java/step/automation/packages/junit/StepAutomationPackageRunWithCategories.java:8]: also added a plan with "category E" just not part of the include/exclude to improve the coverage

**Review by jeromecomte — APPROVED** [review]

**Pull request merged**